postProcessing/Folders/CosmicIntegration/PythonScripts/ClassCOMPAS.py

```diff
@@ -69,9 +71,26 @@ def setCOMPASDCOmask(
         # By default, we mask for BBHs that merge within a Hubble time, assumming
         # the pessimistic CEE prescription (HG donors cannot survive a CEE) and
         # not allowing immediate RLOF post-CEE
+        
         stellar_type_1, stellar_type_2, hubble_flag, dco_seeds = \
             self.get_COMPAS_variables("BSE_Double_Compact_Objects", ["Stellar_Type(1)", "Stellar_Type(2)", "Merges_Hubble_Time", "SEED"])
 
+        if types == "CHE_BBH":
```

Comment:
No strong opinion here but you don't _need_ this condition here.

I guess you're doing this to speed up the function by not calculating the CHE stuff when you don't want to use it for the DCO mask? The alternate argument is that if you calculate it up front now you'll have all of the masks available in the class in case you want to use it later and then won't have to re-run this mask function (you could instead do `self.DCOmask = self.CHE_BBHmask` for example).

But as I say, no strong opinion, either works for me.

Reply:
> I guess you're doing this to speed up the function by not calculating the CHE stuff when you don't want to use it for the DCO mask? 

Actually, no - the check is needed (or some other coding is required).  The che_ms_x flags are not in the output files by default (maybe they should be now that CHE is enabled by default - I may open an issue for that), so get_COMPAS_variables() will fail if it is called and those columns aren't present in the file (we could put exceptions in get_COMPAS_variables(), but we'd still need to deal with those columns not being there).  The check is so that we only try to get those columns if CHE is requested.  The columns may still not be in the file, but there is more chance of them being there if the user is requesting CHE BBHs.  Maybe we should be more defensive and test for all requested columns in get_COMPAS_variables() (via try/except) and return an error on failure...

Reply:
Ahhh gotcha, I hadn't realised there weren't a default, makes sense in that case! We could definitely add some try/excepts, though I find the hdf5 errors are usually fairly clear about what it missing?

Reply:
Oh yeah, the hdf5 errors are clear, but it's an inelegant way of failing.  If we all had lots of time to devote to coding things correctly we'd control the code flow (i.e. trap the error, print an error message, and decide what to do - we might just fail, proceed without CHE, or maybe ask the user what they wanted to do (in this case, probably just fail)).  It's a bit like 'cp' or 'ls' on Linux failing and printing a stack trace because a file doesn't exists - instead of just printing a simple error message. But we don't have lots of time, so as long as people understand what the problem is when it fails it's ok (just inelegant :-)